config/filter.d/exim.conf

```diff
@@ -18,7 +18,7 @@ failregex = ^%(pid)s %(host_info)ssender verify fail for <\S+>: (?:Unknown user|
             ^%(pid)s %(host_info)srejected RCPT [^@]+@\S+: (?:relay not permitted|Sender verify failed|Unknown user|Unrouteable address)\s*$
             ^%(pid)s SMTP protocol synchronization error \([^)]*\): rejected (?:connection from|"\S+") %(host_info)s(?:next )?input=".*"\s*$
             ^%(pid)s SMTP call from \S+ %(host_info)sdropped: too many nonmail commands \(last was "\S+"\)\s*$
-            ^%(pid)s SMTP protocol error in "AUTH \S*(?: \S*)?" %(host_info)sAUTH command used when not advertised\s*$
+            ^%(pid)s SMTP protocol error in "[Aa][Uu][Tt][Hh] \S*(?: \S*)?" %(host_info)sAUTH command used when not advertised\s*$
```

Comment:
Please don't do that, just use `(?:AUTH|auth)` or even case-insensitive `(?:(?i)AUTH)` instead of.

Reply:
`(?:AUTH|auth)` won't match mixed case (`AuTh`), which is also valid SMTP.

`(?:(?i)AUTH)` is ok with me, but it results in the entire expression (not limited by the parentheses) becoming case-insensitive, thus making it less restrictive than it was. If you are ok with that, please confirm and I will change it to this.

Reply:
You are right (I forgot the non-branching `(?i)` PCRE-variant that used in python...). 
Then leave it as is...
Or what do you think about:
```diff
- SMTP protocol error in "AUTH \S*(?: \S*)?" ... AUTH command used when not advertised
+ SMTP protocol error in "[^"]+" ... AUTH command used when not advertised
```
IMHO, the anchor "AUTH command used when not advertised" is precise enough here, and it will prevent by any user data here (because really forign-input).

I'll merge it later if no other objections follow.

Reply:
Sorry for the late reply, but yes, your proposal is probably best. I agree that the final part is probably specific enough. I did some manual tests and ran `fail2ban-testcases -n` and everything seems to work ok, no regressions.

Reply:
BTW only one thing I take care would be potential "injection" like `AUTH login "test"` (with quotes), but currently no time to check this.
If it produces an unescaped output in exim-log, then the expression should be rewritten as more precise...

Reply:
Ouch, they do go through unescaped:

    2017-11-30 06:11:27 SMTP protocol error in "AUTH "LOGIN"" H=localhost (peternowee.com) [127.0.0.1] AUTH command used when not advertised
    2017-11-30 06:12:04 SMTP protocol error in "AUTH LOGIN "test"" H=localhost (peternowee.com) [127.0.0.1] AUTH command used when not advertised

These lines get matched by the current regex `"AUTH \S*(?: \S*)?"`, but not by the new one `"[^"]+"`. The client can enter any long string and Exim will just copy it there, although it does seem to strip trailing spaces. Commands with more than two arguments (e.g. `"AUTH LOGIN " ""`) and just the command `"AUTH"` do not get matched by `"AUTH \S*(?: \S*)?"` either, though these would probably not be used in login attempts anyway.

Perhaps match any string in the middle `".*"` then, and trust the anchors at the start and the end? Here is [a Debuggex using `".*"` in which I let a malicious client (192.0.2.6) try to inject a match for a third-party victim (192.0.2.5)](https://www.debuggex.com/r/tUSSaffQyEfN6pzO). This correctly fails because of the `$` anchoring at the end. I also checked with `fail2ban-testcases`: It sees host 192.0.2.6. Looking at the [Exim source code for these error messages](https://github.com/Exim/exim/blob/exim-4_89+fixes/src/src/smtp_in.c#L2850), I don't expect Exim will produce log lines starting with "SMTP protocol error in" and ending in foreign input.

Reply:
> Perhaps match any string in the middle ".*" ....

Nope. Although I don't think that more complex injection could go through (because of the end-anchor). Also with the second foreign input from `helo` logged here, etc. But...
Just "pure" catch-all is always bad, just because the string can be long (thus it slows down the whole handling). I saw too often how the attackers paralyze the whole service with pair lines affected by such expressions.
But because we are forced, I'll add one but not greedy as alternating fallback (no chance otherwise).